web-ng/src/app/components/layer-list-item/layer-list-item.component.ts

```diff
@@ -88,31 +88,28 @@ export class LayerListItemComponent implements OnInit, OnDestroy {
     this.navigationService.setFeatureId(null);
   }
 
-  onDeleteFeature() {
+  onDeleteLayer() {
     const dialogRef = this.confirmationDialog.open(
       ConfirmationDialogComponent,
       {
         maxWidth: '500px',
         data: {
           title: 'Warning',
           message:
-            'Are you sure you wish to delete this feature? Any associated data including all observations in this feature will be lost. This cannot be undone.',
+            'Are you sure you wish to delete this layer? Any associated data including all features in this layer will be lost. This cannot be undone.',
         },
       }
     );
 
     dialogRef.afterClosed().subscribe(async dialogResult => {
       if (dialogResult) {
-        await this.deleteFeature();
+        await this.deleteLayer();
       }
     });
   }
 
-  async deleteFeature() {
-    if (!this.projectId || !this.featureId) {
-      return;
-    }
-    await this.dataStoreService.deleteFeature(this.projectId, this.featureId);
+  async deleteLayer() {
+    await this.dataStoreService.deleteLayer(this.projectId!, this.layer!.id);
```

Comment:
Please also implement deletion of observations in data store service ( `where('layerId', '==', layerId` ).